app/views/shop/products/_summary.html.haml

```diff
@@ -10,11 +10,16 @@
       %a{"ng-click" => "triggerProductModal()", href: 'javascript:void(0)'}
         %span{"ng-bind" => "::product.name"}
     %p.product-description{ng: {bind: "::product.description", click: "triggerProductModal()", show: "product.description.length"}}
-    .product-producer
-      = t :products_from
-      %span
-        %enterprise-modal
-          %span{"ng-bind" => "::enterprise.name", "ng-class" => "{'red': enterprise.visible == 'hidden'}"}
+    %div{ "ng-switch" => "enterprise.visible" }
+      .product-producer{ "ng-switch-when": "hidden" }
+        = t :products_from
+        %span
+          %span.grey{"ng-bind" => "::enterprise.name"}
```

Comment:
If the "products from" text is grey, maybe we don't need any `span` elements at all in this case?